src/API/posts.js

```diff
@@ -0,0 +1,42 @@
+import { client } from '../utils/fetchClient';
+
+export const getPosts = async (userId) => {
+  const response = await client.get(`/posts?userId=${userId}`);
+
+  return response.data;
+};
+
+export const addPostToServer = async (userId, postData) => {
+  const response = await client.post('/posts', {
+    userId,
+    ...postData,
+  }, {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded'
+    }
+  });
+
+  return response.data;
+};
+
+export const editServerPost = async (postId, postData) => {
+  const response = await client.patch(`/posts/${postId}`, {
+    ...postData,
+  }, {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded'
```

Comment:
The `Content-Type` header is set to `application/x-www-form-urlencoded`. Verify if this is necessary for PATCH requests. JSON is typically used for sending data in PATCH requests.